File: 94-Binary-Tree-Inorder-Traversal.cpp
class Solution {
public:
    vector<int> inorderTraversal(TreeNode* root) {
        if (root == nullptr)
            return {};

        vector<int> res;
        traversal(res, root);
        return res;
    }

private:
    void traversal(vector<int>& res, TreeNode* root) {
        if (root == nullptr)
            return;

        traversal(res, root->left);
        res.push_back(root->val);
        traversal(res, root->right);
    }
};

File: 94-Binary-Tree-Inorder-Traversal.py
class Solution:
    def inorderTraversal(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return []

        res = []
        def traversal(root: TreeNode) -> None:
            if not root:
                return
            traversal(root.left)
            res.append(root.val)
            traversal(root.right)
        
        traversal(root)
        return res
            
